libpod/runtime_pod.go

```diff
@@ -129,3 +130,59 @@ func (r *Runtime) GetLatestPod() (*Pod, error) {
 	}
 	return pods[lastCreatedIndex], nil
 }
+
+// GetRunningPods returns an array of running pods
+func (r *Runtime) GetRunningPods() ([]*Pod, error) {
+	var (
+		cids        []string
+		runningPods []*Pod
+	)
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	if !r.valid {
+		return nil, ErrRuntimeStopped
+	}
+	pods, err := r.state.AllPods()
+	if err != nil {
+		return nil, err
+	}
+	containers, err := r.GetRunningContainers()
+	if err != nil {
+		return nil, err
+	}
+	// Assemble a slice of cids for lookup
+	for _, c := range containers {
+		cids = append(cids, c.ID())
+	}
+	for _, p := range pods {
+		containersByID, err := p.AllContainersByID()
+		if err != nil && (err == ErrPodRemoved || err == ErrNoSuchPod) {
+			return nil, err
+		}
+		for _, c := range containersByID {
+			// if we find one running container, the pod is considered to be running.
+			if util.StringInSlice(c, cids) {
+				runningPods = append(runningPods, p)
+				break
+			}
+		}
+	}
+	return runningPods, nil
+}
+
+// GetPodsByList returns a slice of pods given a string list of wanted pod
+// ids or names
+func (r *Runtime) GetPodsByList(podList []string) ([]*Pod, error) {
```

Comment:
Either add a `GetContainersByList` or move this into cmd/podman - let's avoid adding things we don't have a container equivalent for.

Reply:
done